[OSDOCS-4464]: ccoctl gcp delete specify dir param

jeana-redhat opened the pull request:
Version(s):
4.10+ (see _Additional information_ below)

Issue:
[OSDOCS-4464](https://issues.redhat.com//browse/OSDOCS-4464)

Link to docs preview:
[Deleting GCP resources with the Cloud Credential Operator utility](https://52380--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/uninstalling-cluster-gcp.html#cco-ccoctl-deleting-sts-resources_uninstalling-cluster-gcp)

QE review:
- [ ] QE has approved this change.

Additional information:
Cherrypicks into 4.11 and 4.10 will need to accompany the code backports and be done in tandem with the .z rel notes.
4.12 RN PR: #52381
4.11 RN PR: 
4.10 RN PR: 

Text for .z rel notes:
```
[id="ocp-4-YY-ZZ-notable-technical-changes"]
==== Notable Technical Changes

* With this release, when you xref:../installing/installing_gcp/uninstalling-cluster-gcp.adoc#cco-ccoctl-deleting-sts-resources_uninstalling-cluster-gcp[delete GCP resources with the Cloud Credential Operator utility], you must specify the directory containing the files for the component `CredentialsRequest` objects.
```

abutcher commented on modules/cco-ccoctl-deleting-sts-resources.adoc:
```diff
@@ -54,13 +54,17 @@ ifdef::google-cloud-platform[]
 +
 ** `<name>` matches the name used to originally create and tag the cloud resources.
 ** `<gcp_project_id>` is the GCP project ID in which cloud resources will be deleted.
+** `<path_to_directory_with_list_of_credentials_requests>/credrequests` is the directory containing the files for the component `CredentialsRequest` objects.
```
There's a good chance that the `CredentialsRequests` from installation won't be available so I think we should link to how to extract them for $version from the installation steps.

jianping-shu commented:
@huangmingxia pls. review

huangmingxia commented:
1. No alignment from the typography.
2. --credentials-requests-dir=<path_to_directory_with_list_of_credentials_requests>
   Should be --credentials-requests-dir=<path_to_directory_with_list_of_credentials_requests>/credrequests
3. No note for <path_to_directory_with_list_of_credentials_requests>/credrequests

jeana-redhat commented:
Thanks, @huangmingxia for the quick review! Embarrassed that I didn't notice that alignment issue :facepalm: I have fixed that, and also added `/credrequests` to the path. I am leaving the note off since it is defined immediately above in step 1 :slightly_smiling_face:

aireilly commented:
/label peer-review-in-progress

aireilly commented on modules/cco-ccoctl-deleting-sts-resources.adoc:
```diff
@@ -41,27 +41,13 @@ ifdef::aws-sts[]
-** `<gcp_project_id>` is the GCP project ID in which cloud resources will be deleted.
-endif::google-cloud-platform[]
+<1> `<name>` matches the name used to originally create and tag the cloud resources.
+<2> `<aws_region>` is the AWS region in which cloud resources will be deleted.
```
Nit: use present tense where possible. 

```suggestion
<2> `<aws_region>` is the AWS region where the cloud resources are deleted from.
```

aireilly commented on modules/cco-ccoctl-deleting-sts-resources.adoc:
```diff
@@ -83,6 +69,33 @@ endif::google-cloud-platform[]
+$ ccoctl gcp delete \
+  --name=<name> \ <1>
+  --project=<gcp_project_id> \ <2>
+  --credentials-requests-dir=<path_to_directory_with_list_of_credentials_requests>/credrequests
```
Same comment as above. `<path_to_directory_with_list_of_credentials_requests>/credrequests` seems unnecessarily verbose.

```suggestion
  --credentials-requests-dir=<credentials_requests_dir>
```

aireilly commented on modules/cco-ccoctl-deleting-sts-resources.adoc:
```diff
@@ -83,6 +69,33 @@ endif::google-cloud-platform[]
+  --credentials-requests-dir=<path_to_directory_with_list_of_credentials_requests>/credrequests
+----
++
+<1> `<name>` matches the name used to originally create and tag the cloud resources.
```
```suggestion
<1> `<name>` matches the name that was originally used to create and tag the cloud resources.
```

aireilly commented on modules/cco-ccoctl-deleting-sts-resources.adoc:
```diff
@@ -83,6 +69,33 @@ endif::google-cloud-platform[]
+  quay.io/<path_to>/ocp-release:<version>
+----
++
+<1> `credrequests` is the directory where the list of `CredentialsRequest` objects is stored. This command creates the directory if it does not exist.
```
```suggestion
<1> `<credentials_requests_dir>` is the directory where the list of `CredentialsRequest` objects is stored. This command creates the directory if it does not exist.
```

aireilly reviewed (commented):
Overall looks good! Left a few nit comments, and one for follow up - you should clarify what `<path_to>` refers to.

:)

aireilly commented:
/remove-label peer-review-in-progress
/label peer-review-done
/remove-label peer-review-in-progress
/remove-label peer-review-needed

jeana-redhat commented on modules/cco-ccoctl-deleting-sts-resources.adoc:
```diff
@@ -83,6 +69,33 @@ endif::google-cloud-platform[]
+----
+$ oc adm release extract --credentials-requests \
+  --cloud=gcp \
+  --to=<path_to_directory_with_list_of_credentials_requests>/credrequests \ <1>
```
It is long, but unfortunately it is this way on purpose due to the possibility of user choices when they set this up

jeana-redhat commented:
Ok, I have made some changes that include a couple tweaks in the commands issued. @huangmingxia PTAL to make sure this works as expected :pray:

huangmingxia commented:
Looks good to me, LGTM

jeana-redhat commented:
/cherrypick enterprise-4.

jeana-redhat commented:
/cherrypick enterprise-4.12

bergerhoffer commented:
@jeana-redhat Same question here as the other CCO PR :)

jeana-redhat commented:
I, also, wondered that today. I've checked the last two 4.11.z and 4.10.z releases and do not see any of the related bugs in them, so perhaps next week?

jianping-shu commented:
Here is the info I found.
4.11.z, ticket is [OCPBUGS-2801](https://issues.redhat.com/browse/OCPBUGS-2801) , already merged into 4.11.14
4.10.z, ticket is [OCPBUGS-2802](https://issues.redhat.com/browse/OCPBUGS-2802) , already merged into 4.10.41

jeana-redhat commented:
Starting this cherrypick ahead of merging the RN update to see if it will need manual intervention
/cherrypick enterprise-4.11

jeana-redhat commented:
/cherrypick enterprise-4.10
